src/Weeds3dViewer/Weeds3dViewer.js

```diff
@@ -119,21 +125,114 @@ const Weeds3dViewer = () => {
       type: 'numeric',
       align: 'justify',
       render: (rowData) => {
-        return Math.round((rowData.file_size / 1000000) * 10) / 10 + ' MB';
+        return convertToMB(rowData.file_size) + ' MB';
+      },
+    },
+    {
+      title: 'File Info',
+      align: 'justify',
+      render: (rowData) => {
+        return convertToMB(rowData.file_size) < 25 ? (
+          <Tooltip title="This file size is too small">
+            <Error style={{ color: '#cc4968' }} />
+          </Tooltip>
+        ) : convertToMB(rowData.file_size) > 750 ? (
+          <Tooltip title="This file size is good">
+            <CheckCircle style={{ color: '#2F7C31' }} />
+          </Tooltip>
+        ) : (
+          <Tooltip title="This file size is informational">
+            <Info style={{ color: '#5e7ee0' }} />
+          </Tooltip>
+        );
       },
     },
   ];
 
+  const convertToMB = (size) => {
+    return Math.round((size / 1000000) * 10) / 10;
+  };
+
+  const fetchCodes = async (apikey) => {
+    // calls the site_information API and gets a list of sites
+    await axios({
+      method: 'GET',
+      url: onfarmAPI + '/raw?output=json&table=site_information',
+      headers: {
+        'x-api-key': apikey,
+      },
+      responseType: 'json',
+      timeout: 5000,
+    })
+      //iterates over sites objects and compiles a list of site codes
+      .then((res) => {
+        let listOfCodes = [];
+        const data = res.data;
+        Object.keys(data).forEach((key) => {
+          listOfCodes.push(data[key].code);
+        });
+        return listOfCodes;
+      })
+      // feeds list of site codes to weeds 3d API and gets list of video files, sets state
+      .then(async (res) => {
+        const data = { codes: res };
+        const files = await callAzureFunction(data, '/weeds3d/videos', 'POST', getTokenSilently);
+        setVideos(files.jsonResponse.files);
+        setLoading(false);
+
+        return files.jsonResponse.files;
+      })
+      //iterates over files and compiles a parameter string of all unique codes, then feeds parameter
+      //string to dates API to get year and affliation information
+      .then(async (res) => {
+        let param = '';
+        res.forEach((video) => {
+          const code = video.code;
+          if (!param.includes(code)) param = param.concat(code + ',');
+        });
+
+        const codeInfo = await axios({
+          method: 'GET',
+          url: onfarmAPI + '/dates?code=' + param,
```

Comment:
why are you calling the dates endpoint here? you already have affiliations and years from site info right? you can make yearsAndAffiliationsDict after calling site info

Reply:
I don't know why I completely forgot about that. no need for another API call